patches/api/0457-Add-ShulkerDuplicateEvent.patch

```diff
@@ -0,0 +1,79 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Chase Henderson <henderson.chase@gmail.com>
+Date: Fri, 5 Jan 2024 03:50:10 -0500
+Subject: [PATCH] Add ShulkerDuplicateEvent
+
+
+diff --git a/src/main/java/io/papermc/paper/event/entity/ShulkerDuplicateEvent.java b/src/main/java/io/papermc/paper/event/entity/ShulkerDuplicateEvent.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..eb32542031088b1735fd0cc90fe350d0016b3622
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/event/entity/ShulkerDuplicateEvent.java
+@@ -0,0 +1,67 @@
++package io.papermc.paper.event.entity;
++
++import org.bukkit.entity.Shulker;
++import org.bukkit.event.Cancellable;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.entity.EntityEvent;
++import org.jetbrains.annotations.ApiStatus;
++import org.jetbrains.annotations.NotNull;
++
++/**
++ * Is called when a {@link org.bukkit.entity.Shulker} duplicates.
++ */
++public class ShulkerDuplicateEvent extends EntityEvent implements Cancellable {
++
++    private static final HandlerList HANDLER_LIST = new HandlerList();
++
++    private final Shulker parent;
++    private boolean cancelled;
++
++    @ApiStatus.Internal
++    public ShulkerDuplicateEvent(@NotNull Shulker child, @NotNull Shulker parent) {
++        super(child);
++        this.parent = parent;
++    }
++
++    /**
++     * Gets the shulker that is duplicating.
++     *
++     * @return The Child
++     */
```

Comment:
I would further clarify that this is the new entity being created, and that it does not exist in the world yet.